app/coffeescript/components/mixins/distance_conversion.coffee

```diff
@@ -0,0 +1,31 @@
+define [ ], () ->
+
+  distanceConversion = ->
+    meterToKilometer = 1000
+    kmRatio = 0.62137
+
+    @_metersToKilometers = (meters) ->
+      return 0 unless @_distanceParamDefined(meters)
+      (meters / meterToKilometer) * kmRatio
+
+    @_milesToMeters = (miles) ->
+      return 0 unless @_distanceParamDefined(miles)
+      (miles / kmRatio) * meterToKilometer
+
+    @_metersToMiles = (meters) ->
+      return 0 unless @_distanceParamDefined(meters)
+      (meters / meterToKilometer) * kmRatio
+
+    @_distanceParamDefined = (p) ->
+      typeof p != 'undefined' && p && p != ''
```

Comment:
please get rid of this fn, with coffeescript to check if foo is defined you just do foo?